server.js

```diff
@@ -985,6 +985,46 @@ routes['join_room'] = function (user, data) {
     room.sync(user);
 }
 
+routes['search_youtube'] = function (user, data) {
+	if(user.room === null) return;
+	var room = user.room;
+
+	query = data.query.trim();
+
+	if (typeof query !== "string" || query.length < 1 || query.length > SEARCH_YOUTUBE_MAX_CHARACTERS) {
+		user.send("search_message", {data: "Search must be between 1 and 100 characters"});
+	}
+	
+        var host = "www.googleapis.com";
+        var path = "/youtube/v3/search?maxResults=5&part=snippet&q=" +
+            	encodeURIComponent(query) +
+            	"&key=" +
+            	YOUTUBE_API_KEY;
+
+	var results = [];
+
+        http_get(host, path, function (body, status_code) {
+		try {
+        		var data = JSON.parse(body);
+			for(var i = 0; i < data.items.length; i++) {
+				var item = data.items[i];
+                       		results.push({
+					id: item.id.videoId, 
+					title: item.snippet.title,
+					thumbnail: item.snippet.thumbnails.default.url
+				});
+                	}
+    			user.send("search_results", {data: results});
+		}
+		catch(error) {
+                	debug("Search error: " + error);
+                	user.send("search_message", {data: "The Youtube search API may be down, please try again."});
```

Comment:
This is highly unlikely. Please change the message to:
Sorry, there was a problem searching. Please try again later.